docs/theming.md

```diff
@@ -0,0 +1,299 @@
+
+    ```js
+    window.ShadyCSS.CustomStyleInterface.addCustomStyle(document.querySelector('style.document-style'));
+    ```
+
+    This code assumes a single document-level style element. If you have more than one, you'll need to modify the code.
+    See []()
+
+With these steps, the polyfills provide limited support for CSS custom properties. 
+See [Limitations](https://github.com/webcomponents/polyfills/tree/master/packages/shadycss#limitations)
+in the ShadyCSS README for details.
+
+## Colors
+
+Material Design theming uses two main colors: **primary** and **secondary**. The primary color is used
+throughout most of the application and components, as the main color for your application. The secondary color is used
+for floating action buttons and other interactive elements, serving as visual contrast to the primary.
+
+In addition to the primary and secondary colors, Material Theming also defines a _background_ color for the app, 
+as well as a _surface_ color, which is used used as a background in components.
+
+Finally, Material Theming has a number of text colors, which are used for rendering text and other shapes on top of the primary,
+secondary and background colors. These are specified as either dark or light, in order to provide sufficient contrast to
+what's behind them, and have
+[different levels of opacity depending on usage](https://material.io/design/color/the-color-system.html#):
+- Primary, used for most text.
+- Secondary, used for text which is lower in the visual hierarchy.
+- Hint, used for text hints (such as those in text fields and labels).
+- Disabled, used for text in disabled components and content.
+- Icon, used for icons.
+- On-surface, used for text that is on top of a surface background.
+- On-secondary, used for text that is on top of a secondary background.
+- On-primary, used for text that is on top of a primary background.
+
+### Color properties
+
+These custom properties used by MWC set the main colors for an application:
+
+| Custom property               | Description                                 |
+| ----------------------------- | ------------------------------------------- |
+| `--mdc-theme-primary`         | The theme primary color.                    |
+| `--mdc-theme-secondary`       | The theme secondary color.                  |
+| `--mdc-theme-surface`         | The theme surface color.                    |
+| `--mdc-theme-background`      | The theme background color.                 |
+
+When you set the main colors, you'll also have to set contrasting colors for text and iconography:
+
+| Custom property                               | Description                                                                |
+| --------------------------------------------- | -------------------------------------------------------------------------- |
+| `--mdc-theme-on-primary`                      | Text and icons on top of a theme primary color background.                   |
```

Comment:
I'm torn on this. Adding literal defaults here duplicates information in the Material design docs, and the information would be repeated for every component, and if the default changes we'd have to update them all.

Reply:
Hm, I do agree, though we do this already in our [CSS custom property](https://github.com/material-components/material-components-web-components/tree/master/packages/button#css-custom-properties) docs per component. @e111077 would have more opinions on it.

Reply:
Seeing as this is an overview explaining theming, I think we can skip default values in this case. I feel like it's a bit more-pertinent when you're dealing with an individual element.

Reply:
It's not really per component though. If I'm changing the primary theme for everything, I should be concerned about the on-primary theme since that will cascade across elements.

```css
html {
  /* this is ok because on-primary is white by default */
  --mdc-theme-primary: blue; 
}

html {
  /* not good by default! I need to set on-primary to black or something */
  --mdc-theme-primary: yellow; 
  --mdc-theme-on-primary: black;
}
```

We could take the approach to always override, but if we do that for every single text custom property, it might be a bit overboard since a lot of them won't need to be changed from their default white/black values.

Reply:
Good point. TBH, I'm not too opinionated on having the default colors, but if we do have them, it should follow the same format of screenshot - value as other docs.

Though the point brought up about doc maintenance sounds like some infra we should work on >.<

Reply:
Yeah, I originally left the color swatches out of the usage docs I've been working on because it seemed like it would result in an inordinate number of images to tote around. But it is much easier to understand than looking at a number...

Reply:
Luckily most of them already exist somewhere in the project directory so you shouldn't have to really have to generate most of them. If you do:

* open about:blank in your browser
* make a div that is 6x6 px
* set the background-color to the correct color
* move chrome window to 2x dpi screen (macbook pro monitor)
* open chrome dev tools
* highlight div
* press command+shift+P
* type "Capture Node Screenshot" and hit enter
* you should have a perfectly-sized 12x12 pixel color screenshot

Reply:
Dumb question... Is there a reason I shouldn't just make it 12x12 in the first place?

I ask this because, given my current setup, step 4 "move chrome window to 2x dpi screen" involves the following steps:
- unplug laptop from external monitor and headset.
- take laptop to another room.
- open laptop, take screenshot, reverse previous steps. 
:joy:

Reply:
tbh that can work since it's just a pixel of a color. This is more important when screenshotting entire components.